sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -454,6 +455,154 @@ describe(
       );
     });
 
+    it("Should not create an e-Cert record for student when the maximum lifetime CSLP amount is less than the sum of latest CSLP balance and the disbursed amount.", async () => {
+      // Arrange
+
+      // Student with valid SIN.
+      const student = await saveFakeStudent(db.dataSource);
+      // Valid MSFAA Number.
+      const msfaaNumber = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.Signed,
+            msfaaInitialValues: {
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
+
+      // Update CSLP balance for the student.
+      await db.studentLoanBalance.insert(
+        createFakeStudentLoanBalance(
+          { student },
+          {
+            initialValues: {
+              balanceDate: "2023-11-30",
+              cslBalance: 10000,
+            },
+          },
+        ),
+      );
+
+      // Student application eligible for e-Cert.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student, msfaaNumber },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Completed,
+          currentAssessmentInitialValues: {
+            assessmentData: { weeks: 5 } as Assessment,
+            assessmentDate: new Date(),
+          },
```

Comment:
As discussed, having the workflow data defined with some default value was better for part time.

Reply:
> As discussed, having the workflow data defined with some default value was better for part time.

Is it about having the workflow data defined in the factory?
If yes, I would like to get a benefit, from a factory perspective, to have the workflow data set. I saw at least one place along the E2Es where specific E2E scanrios were wrongly set inside the factory, reducing its generic idea.

Reply:
The dmnValues that are defined in the workflowData for part-time are used to define a value for the lifetimeMaximumCSLP. If that value is not set or set to 0, we should define the values in other test cases for part-time ecert disbursement explicitly, else it will fail the test case by making the disbursement blocked. 
Having it defaulted to a value such as 10000, makes the part-time ecert generation go through.

Reply:
Yeah, I got the idea and the benefits of this scenario. As we discussed we should plan to have `workflowData` built outside the method that it is right now and make it more specialized. For this PR we are good.